docs/proposals/fluentd-limit-requests-to-apiserver.md

```diff
@@ -0,0 +1,63 @@
+# Modifications to Fluentd Metadata Retrieval Proposal
+
+##Motivation
+In the current design (v3.4) of the EFK stack, there is a Fluentd pod deployed to each Openshift node. Fluentd is responsible for reading container logs, adding pod metadata (e.g. labels, namespace) to individual log entries, and sending the logs to the Elasticsearch cluster. The pod metadata is sourced by watching the Openshift API server for Pod information via the [Kubernetes Metadata Filter](https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter)[2]. Testing at larger scales of Openshift clusters has shown issues where Fluentd is unable to connect to the API server because it is limited by the **maxRequestsInFlight** configuration setting [1].  This results in logs not being shipped to the aggregator and thus being unavailable.
```

Comment:
Looking at the REST client that https://github.com/abonas/kubeclient uses, https://github.com/rest-client/rest-client, shows up some issues around connection reuse. I'm not a big enough Ruby user to know in detail, but perhaps @abonas could comment or know someone who could help out? It might require changes to https://github.com/rest-client/rest-client to support this - seeing as this client is used in manageiq too that might very well be worth doing.